src/index.js

```diff
@@ -214,53 +203,74 @@ export default function undoable (reducer, rawConfig = {}) {
     jumpType: rawConfig.jumpType || ActionTypes.JUMP,
     clearHistoryType: rawConfig.clearHistoryType || ActionTypes.CLEAR_HISTORY
   }
-  config.history = rawConfig.initialHistory || createHistory(config.initialState || reducer(undefined, {}))
```

Comment:
Setting the history here won't work, because when we reset it later we expect it to reset to the store's initialState.. So I think we should postpone this line to where the reducer is actually called for the first time.